cmake/macros/Public.cmake

```
@@ -1202,6 +1202,20 @@ function(pxr_toplevel_epilogue)
             )
         endif()
 
+        # When building a monolithic library (static or shared) we want all API functions to be
+        # exported. So add FOO_EXPORTS=1 for every library in PXR_OBJECT_LIBS,
+        # where FOO is the uppercase version of the library name, to every
+        # library in PXR_OBJECT_LIBS.
+        set(exports "")
+        foreach(lib ${PXR_OBJECT_LIBS})
+            string(TOUPPER ${lib} uppercaseName)
+            list(APPEND exports "${uppercaseName}_EXPORTS=1")
+        endforeach()
+
```

```suggestion

if (TARGET python)
    # The boost python target uses a different export macro so
    # add that as well.
    list(APPEND exports "PXR_BOOST_PYTHON_SOURCE=1")
endif()

```

When we tested this fix on our side, we noticed there were still a few thousand warnings from the Python binding code in `pxr/external/boost/python`. This library doesn't follow the same `<FOO>_EXPORTS` pattern, so we needed to add this in to get rid of those warnings.

Could you give this a try on your side to confirm this works? If so, would you mind updating your PR to include this?

---

Added. Thanks for checking.